addons/binding/org.openhab.binding.kostalinverter/src/main/java/org/openhab/binding/internal/kostal/inverter/ThirdGeneration/ThirdGenerationHandler.java

```diff
@@ -0,0 +1,767 @@
+                double valueAsDouble = newValueObject.get("value").getAsDouble();
+                ThirdGenerationChannelMappingToWebApi channel = channelList.get(moduleName).stream()
+                        .filter(c -> c.moduleId.equals(moduleName) && c.processdataId.equals(valueId)).findFirst()
+                        .get();
+                updateChannelValue(channel.channelUID, channel.dataType, valueAsDouble);
+            }
+        }
+    }
+
+    /**
+     * Update the channel to the given value.
+     * The value is set to the matching data (SITypes etc)
+     *
+     * @param channeluid Channel to update
+     * @param dataType   target data type
+     * @param value      value
+     */
+    private void updateChannelValue(String channeluid, ThirdGenerationChannelDatatypes dataType, Double value) {
+        switch (dataType) {
+            case Integer: {
+                updateState(channeluid, new DecimalType(value.longValue()));
+                break;
+            }
+            case Percent: {
+                updateState(channeluid,
+                        new QuantityType<javax.measure.quantity.Dimensionless>(value, SmartHomeUnits.PERCENT));
+                break;
+            }
+            case KiloGram: {
+                updateState(channeluid, new QuantityType<javax.measure.quantity.Mass>(value / 1000, SIUnits.KILOGRAM));
+                break;
+            }
+            case Seconds: {
+                updateState(channeluid, new QuantityType<javax.measure.quantity.Time>(value, SmartHomeUnits.SECOND));
+                break;
+            }
+            case KiloWattHour: {
+                updateState(channeluid,
+                        new QuantityType<javax.measure.quantity.Energy>(value / 1000, SmartHomeUnits.KILOWATT_HOUR));
+                break;
+            }
+            case Watt: {
+                updateState(channeluid, new QuantityType<javax.measure.quantity.Power>(value, SmartHomeUnits.WATT));
+                break;
+            }
+            case Ampere: {
+                updateState(channeluid,
+                        new QuantityType<javax.measure.quantity.ElectricCurrent>(value, SmartHomeUnits.AMPERE));
+                break;
+            }
+            case AmpereHour: {
+                /*
+                 * Ampere hours are not supported by ESH, but 1 AH is equal tp 3600 coulomb...
+                 */
+                updateState(channeluid,
+                        new QuantityType<javax.measure.quantity.ElectricCharge>(value * 3600, SmartHomeUnits.COULOMB));
+                break;
+            }
+            case Volt: {
+                updateState(channeluid,
+                        new QuantityType<javax.measure.quantity.ElectricPotential>(value, SmartHomeUnits.VOLT));
+                break;
+            }
+            default: {
+                /*
+                 * unknown datatype
+                 */
+                logger.error(dataType + " not known!");
```

Comment:
Don't log to `error`, it's reserved for system wide problems. This is should be logged to debug.

Reply:
Will be in the next commit